source/utils.ts

```diff
@@ -36,7 +36,18 @@ export function typecheck(context: Rule.RuleContext) {
   const typeChecker = service.program.getTypeChecker();
 
   const getTSType = (node: es.Node) => {
-    const tsNode = nodeMap.get(node);
+    let tsNode = nodeMap.get(node);
+
+    if (
+      ts.isFunctionDeclaration(tsNode) ||
+      ts.isMethodDeclaration(tsNode) ||
+      ts.isFunctionExpression(tsNode) ||
+      ts.isCallSignatureDeclaration(tsNode) ||
+      ts.isMethodSignature(tsNode)
+    ) {
+      tsNode = tsNode.type;
+    }
```

Comment:
I don't think `getTSType` should automatically 'unpack' a function's return type. It's doing so limits how it can be used. What if I want to use it in another rule to get the type of the function? Instead, I think you should 'unpack' the return type in your rule's report function. You might want to add another report function that's called for function declarations, method declarations, etc. - whatever your selectors are going to match.

Reply:
I have reverted the util function its original version and created local function to resolve the problem. 
However, the functions in utils are kind of chained down and required me to almost not using the function that utils provide.
Unless I introduce extra parameter like `useType` and I can pass it in from `canBeObservable()` like `couldBeObservable: (node: es.Node, useType: boolean = false)`. That way, the decision to 'unpack' is up to the caller. 
What do you think?